sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/FlatMapElements.java

```diff
@@ -45,16 +45,13 @@
     extends PTransform<PCollection<? extends InputT>, PCollection<OutputT>> {
   private final transient @Nullable TypeDescriptor<InputT> inputType;
   private final transient @Nullable TypeDescriptor<OutputT> outputType;
-  private final transient @Nullable Object originalFnForDisplayData;
-  private final @Nullable Contextful<Fn<InputT, Iterable<OutputT>>> fn;
+  private final @Nullable Object fn;
 
   private FlatMapElements(
-      @Nullable Contextful<Fn<InputT, Iterable<OutputT>>> fn,
-      @Nullable Object originalFnForDisplayData,
+      @Nullable Object fn,
```

Comment:
Shouldn't this still be an InferableFunction? Or is the goal to accept either InferableFunction or Contextful? (If the latter, is this too painful to encode in the type system?)

Reply:
You're correct, the type system was too painful to specify `OR` between `ProcessFunction` and `Contextful`.

Note that the constructor is private so this detail isn't exposed to users.